src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2441,21 +2445,21 @@ protected virtual bool MarkMethods (TypeDefinition type, in DependencyInfo reaso
 		void MarkMethodCollection (IList<MethodDefinition> methods, in DependencyInfo reason, IMemberDefinition sourceLocationMember)
 		{
 			foreach (MethodDefinition method in methods)
-				MarkMethod (method, reason, sourceLocationMember);
+				MarkMethod (method, reason, new MessageOrigin (sourceLocationMember));
 		}
 
 		protected internal void MarkIndirectlyCalledMethod (MethodDefinition method, in DependencyInfo reason, IMemberDefinition sourceLocationMember)
 		{
-			MarkMethod (method, reason, sourceLocationMember);
+			MarkMethod (method, reason, new MessageOrigin (sourceLocationMember));
 			Annotations.MarkIndirectlyCalledMethod (method);
 		}
 
-		protected virtual MethodDefinition MarkMethod (MethodReference reference, DependencyInfo reason, IMemberDefinition sourceLocationMember)
+		protected virtual MethodDefinition MarkMethod (MethodReference reference, DependencyInfo reason, MessageOrigin? origin)
```

Comment:
Do we need this change to take `MessageOrigin`? It seems we only ever use the `MemberDefinition` from it, so the previous state which passed the member directly seem to work the same way.

Reply:
This was changed so that we don't lose the line information of the warnings that originate from `MarkInstruction`: https://github.com/mono/linker/blob/e7af1b8756a5845ce46909980d64f2b4c2702100/src/linker/Linker.Steps/MarkStep.cs#L3078
 We can change this back to IMemberDefinition if willing to loose that.